### PR TITLE
Add temporary riddles, adjust positioning of UI elements

### PR DESCRIPTION
**Pull request by travis-bittel**

### CODE REVIEW

**Comment by travis-bittel** [comment]
Merge after syllabus riddles are merged into main

File: Gaming The Syllabus/Assets/AnswerSubmissionManager.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/// <para>Setup: Attach to submit button and set onClick to invoke <see cref="SubmitAnswer"/></para>
+/// </summary>
+public class AnswerSubmissionManager : Singleton<AnswerSubmissionManager>
+{
+    private List<IRiddleAnswerUIElement> selectedAnswers = new();
+
+    private void Awake()
+    {
+        InitializeSingleton();
+    }
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/AnswerSubmissionManager.cs:16]: Good call on making this a singleton. I've caught myself a few times forgetting to extend them on classes I create where they are very much needed.

File: Gaming The Syllabus/Assets/Scripts/Riddles/Riddle.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    public RiddleAnswer[] Answers
+    {
+        get => _answers.ToArray();
+    }
+
+    public RiddleAnswer[] CorrectAnswers
+    {
+        get => _answers.FindAll((answer) => answer.IsCorrect).ToArray();
+    }
+
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/Riddle.cs:25]: This is a good reminder to myself to utilize arrow functions more. It definitely keeps things more compact and concise when used appropriately. Great work!

File: Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public static bool AttemptAnswer(Riddle riddle, RiddleAnswer answer)
+    {
+        // TODO: Make this give the player the ability reward
+        // TODO: Make this punish the player for an incorrect answer
+        // TODO: Potentially properly support "select all correct answers" questions depending on if the client wants them
+        Debug.Log($"Attempted answer to riddle {riddle.Question} with answer {answer.Answer} and result {riddle.CorrectAnswers.Contains(answer)}");
+        return riddle.CorrectAnswers.Contains(answer);
+    }
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs:31]: I definitely wanna test this with my PlayerAbilities implementation. The GiveAbility method inside AbilityController.cs should be able to handle this just fine.

File: Gaming The Syllabus/Assets/Scripts/Riddles/UI/SyllabusRiddleUIManager.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    private static void CreateAnswerElement(GameObject prefab, Vector2 position, Riddle riddle, RiddleAnswer answer)
+    {
+        GameObject answerElement = Instantiate(prefab, _instance.transform);
+        answerElement.GetComponent<IRiddleAnswerUIElement>().Initialize(riddle, answer);
+
+        RectTransform answerElementTransform = answerElement.GetComponent<RectTransform>();
+        answerElementTransform.anchoredPosition = position;
+    }
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/UI/SyllabusRiddleUIManager.cs:72]: Good use of instantiating prefabs here.

File: Gaming The Syllabus/Assets/AnswerSubmissionManager.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        else if (_instance.selectedAnswers.Contains(element))
+        {
+            _instance.selectedAnswers.Remove(element);
+        }
+    }
+
+    public void SubmitAnswer()
+    {
+        selectedAnswers.ForEach((answerElement) => SyllabusRiddleManager.AttemptAnswer(answerElement.AssociatedRiddle, answerElement.GetAnswer()));
+    }
```
Comment (Isildur512) [inline Gaming The Syllabus/Assets/AnswerSubmissionManager.cs:38]: Interesting way to get this all in one line. It keeps the code very readable.

**Pull request merged**